nameko_amqp_retry/backoff.py

```diff
@@ -65,7 +69,22 @@ def get_next_expiration(self, message, backoff_exchange_name):
             randomised = int(random.gauss(expiration, self.random_sigma))
             group_size = self.random_sigma / self.random_groups_per_sigma
             expiration = round_to_nearest(randomised, interval=group_size)
-        return expiration
+
+        # Prevent any negative values created by randomness
+        expiration = abs(expiration)
+
+        # store calculation results on self.
+        self.next_expiration = expiration
+        self.total_attempts = total_attempts
+
+    def __str__(self):
+        return '{}({})'.format(
+            type(self),
+            'retry #{} in {}ms'.format(
+                self.total_attempts + 1, self.next_expiration)
+            if self.next_expiration is not None
+            else 'not-yet-calculated'
```

Comment:
Let's drop the word "calculated" if we're renaming the function. Maybe "uninitialised"?